augur/parse.py

```diff
@@ -68,6 +68,63 @@ def prettify(x, trim=0, camelCase=False, etal=None, removeComma=False):
     return res
 
 
+def parse_sequence(sequence, fields, strain_key="strain", separator="|", prettify_fields=None, fix_dates=None):
+    """Parse a single sequence record into a sequence record and associated metadata.
+
+    Parameters
+    ----------
+    sequence : Bio.SeqRecord.SeqRecord
+        a BioPython sequence record to parse with metadata stored in its description field.
+
+    fields : list or tuple
+        a list of names for fields expected in the given record's description.
+
+    strain_key : str
+        name of the field to use as the given sequence's unique id
+
+    separator : str
+        delimiter to split record description by.
+
+    prettify_fields : list or tuple
+        a list of field names for which the values in those fields should be prettified.
+
+    fix_dates : str
+        parse "date" field into the requested canonical format ("dayfirst" or "monthfirst").
+
+    Returns
+    -------
+    Bio.SeqRecord.SeqRecord :
+        a BioPython sequence record with the given sequence's name as the record
+        id and all other metadata stripped.
+
+    dict :
+        metadata associated with the given record indexed by the given field names.
+    """
+    sequence_fields = map(str.strip, sequence.description.split(separator))
+    metadata = dict(zip(fields, sequence_fields))
+
+    tmp_name = metadata[strain_key].translate(forbidden_chactacters)
+    sequence.name = sequence.id = tmp_name
+    sequence.description = ''
+
+    if prettify_fields:
+        for field in metadata.keys() & prettify_fields:
```

Comment:
What version of python are you using? I'm on 3.7.4 and I get a TypeError when I try to use the `&` operator to concatenate(?) a `dict.keys()` and a `list` or `tuple`

```
TypeError: unsupported operand type(s) for +: 'dict_keys' and 'list'
```

Reply:
The `&` operator between `dict_keys` and a list should work for Python 3.6+. The [`dict_keys` are set-like objects and support the same operators as sets](https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects). The `+` operator doesn't work for sets either though.